src/services/web3.js

```diff
@@ -573,8 +574,76 @@ function sendTransaction(
   }
 }
 
+function getGasPrice(network: Network): Promise<BigNumber> {
+  const {
+    rpcaddr,
+    rpcport,
+    ssl,
+  } = network
+
+  const baseProps = {
+    rpcaddr,
+    rpcport,
+    ssl,
+  }
+
+  return jibrelContractsApi.eth.getGasPrice(baseProps)
+}
+
+function getNonce(network: Network, address: Address): Promise<number> {
+  const {
+    rpcaddr,
+    rpcport,
+    ssl,
+  } = network
+
+  const baseProps = {
+    rpcaddr,
+    rpcport,
+    ssl,
+    address,
+  }
+
+  return jibrelContractsApi.eth.getNonce(baseProps)
+}
+
+function estimateGas(
+  network: Network,
+  assetAddress: AssetAddress,
+  props: SendTransactionProps,
+): Promise<number> {
+  const {
+    rpcaddr,
+    rpcport,
+    ssl,
+  }: Network = network
+
+  if (checkETH(assetAddress)) {
+    return jibrelContractsApi.eth.estimateGas({
+      ssl,
+      rpcaddr,
+      rpcport,
+      to: props.to,
+      value: props.value,
+    })
+  } else {
+    return jibrelContractsApi.contracts.erc20.estimateGas({
+      ssl,
+      rpcaddr,
+      rpcport,
+      contractAddress: assetAddress,
+      privateKey: props.privateKey,
+      method: 'transfer',
+      args: [props.to, props.value],
+    })
```

Comment:
we shouldn't pass privateKey to estimateGas call

Reply:
`from` parameter is necessary to estimate gas

Reply:
Now, it is required https://github.com/jibrelnetwork/jibrel-contracts-jsapi

contracts.erc20.estimateGas(props)
Returns Promise that will be resolved with estimate gas value.

props
rpcaddr: RPC address of Ethereum node to connect on
rpcport: RPC port of Ethereum node to connect on
contractAddress: Contract address
privateKey: Private key (64 hex symbols, without '0x' prefix)
method: Method name
args: Method arguments
ssl: (optional) Defines using of ssl for connection or not

Reply:
but you shouldn't use this function.
instead of it, it should be `call` of contract method
[check example in off docs](https://github.com/ethereum/wiki/wiki/JavaScript-API#contract-methods)

Reply:
Will be fixed in the another task JWWEB-449